build/utils.ts

```diff
@@ -311,29 +311,30 @@ export function makeTOC(doc) {
 }
 
 export function findPostFileBySlug(slug: string): string | null {
-  if (BLOG_ROOT) {
-    try {
-      const { stdout, stderr, status } = spawnSync(rgPath, [
-        "-il",
-        `slug: ${slug}`,
-        BLOG_ROOT,
-      ]);
-      if (status === 0) {
-        const file = stdout.toString("utf-8").split("\n")[0];
-        return file;
+  if (!BLOG_ROOT) {
+    console.warn("'BLOG_ROOT' not set in .env file");
+    return null;
+  }
+
+  try {
+    const { stdout, stderr, status } = spawnSync(rgPath, [
+      "-il",
+      `slug: ${slug}`,
+      BLOG_ROOT,
+    ]);
+    if (status === 0) {
+      const file = stdout.toString("utf-8").split("\n")[0];
+      return file;
+    } else {
```

Comment:
maybe we could remove the 'else' here (when the `if` condition is true, the function would return)